packages/desktop/src/webview.rs

```diff
@@ -232,10 +233,93 @@ impl WebviewInstance {
 
         let file_drop_handler = {
             to_owned![file_hover];
-            move |evt| {
+
+            #[cfg(windows)]
+            to_owned![dom];
+
+            move |evt: DragDropEvent| {
                 // Update the most recent file drop event - when the event comes in from the webview we can use the
                 // most recent event to build a new event with the files in it.
+                #[cfg(not(windows))]
                 file_hover.set(evt);
+
+                // Windows webview blocks HTML-native events when the drop handler is provided.
+                // The problem is that the HTML-native events don't provide the file, so we need this.
+                // Solution: this glue code to mimic drag drop events.
+                #[cfg(windows)]
+                {
+                    use dioxus_html::prelude::eval;
+                    file_hover.set(evt.clone());
+
```

Comment:
Instead of wrapping the vdom in a RefCell, can we just make a new message type on WindowUserEvent that executes this with an &mut access to dom? AFAIK `eval` is not sync anyways